# Exploring eBay Car Sales Data

We will work with a dataset of used cars from eBay Kleinanzeigen, a classifieds section of the German eBay website. The aim of this project is to clean the data and analyze the included used car listings.


In [3]:
import pandas as pd
import numpy as np

autos = pd.read_csv('autos.csv', encoding = 'latin1')

In [4]:
autos.head()

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

In [5]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   dateCrawled          50000 non-null  object
 1   name                 50000 non-null  object
 2   seller               50000 non-null  object
 3   offerType            50000 non-null  object
 4   price                50000 non-null  object
 5   abtest               50000 non-null  object
 6   vehicleType          44905 non-null  object
 7   yearOfRegistration   50000 non-null  int64 
 8   gearbox              47320 non-null  object
 9   powerPS              50000 non-null  int64 
 10  model                47242 non-null  object
 11  odometer             50000 non-null  object
 12  monthOfRegistration  50000 non-null  int64 
 13  fuelType             45518 non-null  object
 14  brand                50000 non-null  object
 15  notRepairedDamage    40171 non-null  object
 16  date

DTYPES are Integers or Objects. | **vehicleType, gearbox, model, fuelType notRepairedDamage** show Nan values. 

## Data Visualization

In [8]:
column_names = autos.columns
column_names

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [9]:
mapping_dict = {'dateCrawled':'date_crawled', 'name': 'name', 'seller': 'seller', 'offerType': 'offer_type', 'price': 'price', 'abtest':'abtest',
       'vehicleType':'vehicle_type', 'yearOfRegistration':'year_of_registration', 'gearbox':'gearbox', 'powerPS':'power_ps', 'model':'model',
       'odometer':'odometer', 'monthOfRegistration':'month_of_registration', 'fuelType':'fuel_type', 'brand':'brand',
       'notRepairedDamage':'not_repaired_damage', 'dateCreated':'ad_created', 'nrOfPictures':'nr_of_pictures', 'postalCode':'postal_code',
       'lastSeen':'last_seen'} # correct column names
column_names = column_names.map(mapping_dict)

In [10]:
autos.columns = column_names
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price   abtest vehicle_type  year_of_registration  \
0  privat    Angebot  $5,000  control          bus                  2004   
1  privat    Angebot  $8,500  control    limousine                  1997   
2  privat    Angebot  $8,990     test    limousine                  2009   
3  privat    Angebot  $4,350  control   kleinwagen                  2007   
4  privat    Angebot  $1,350     test        kombi                  2003   

     gearbox  power_ps   model   odometer  month_of_registration fuel_type  \
0    manuell       158  andere  150,000km                      3       lpg   
1  automatik       286     7er  150,000km                      6    benzin   
2    manuell       102    golf   70,000km                      7    benzin   
3  automatik        71  fortwo   70,000km                      6    benzin   
4    manuell         0   focus  150,000km                      7    benzin   

        brand not_repaired_damage           ad_created  nr_of_pictures  \
0     peugeot                nein  2016-03-26 00:00:00               0   
1         bmw                nein  2016-04-04 00:00:00               0   
2  volkswagen                nein  2016-03-26 00:00:00               0   
3       smart                nein  2016-03-12 00:00:00               0   
4        ford                nein  2016-04-01 00:00:00               0   

   postal_code            last_seen  
0        79588  2016-04-06 06:45:54  
1        71034  2016-04-06 14:45:08  
2        35394  2016-04-06 20:15:37  
3        33729  2016-03-15 03:16:28  
4        39218  2016-04-01 14:38:50

Above, I changed the column names to the snakecase standard to make it easier to visualize the data.

## Data Cleaning



First, looking for: 
* Text columns where all or almost all values are the same. These can often be dropped as they don't have useful information for analysis.
* Examples of numeric data stored as text which can be cleaned and converted.

In [13]:
autos.describe(include = 'all')

date_crawled         name  seller offer_type  price abtest  \
count                 50000        50000   50000      50000  50000  50000   
unique                48213        38754       2          2   2357      2   
top     2016-04-02 11:37:04  Ford_Fiesta  privat    Angebot     $0   test   
freq                      3           78   49999      49999   1421  25756   
mean                    NaN          NaN     NaN        NaN    NaN    NaN   
std                     NaN          NaN     NaN        NaN    NaN    NaN   
min                     NaN          NaN     NaN        NaN    NaN    NaN   
25%                     NaN          NaN     NaN        NaN    NaN    NaN   
50%                     NaN          NaN     NaN        NaN    NaN    NaN   
75%                     NaN          NaN     NaN        NaN    NaN    NaN   
max                     NaN          NaN     NaN        NaN    NaN    NaN   

       vehicle_type  year_of_registration  gearbox      power_ps  model  \
count         44905          50000.000000    47320  50000.000000  47242   
unique            8                   NaN        2           NaN    245   
top       limousine                   NaN  manuell           NaN   golf   
freq          12859                   NaN    36993           NaN   4024   
mean            NaN           2005.073280      NaN    116.355920    NaN   
std             NaN            105.712813      NaN    209.216627    NaN   
min             NaN           1000.000000      NaN      0.000000    NaN   
25%             NaN           1999.000000      NaN     70.000000    NaN   
50%             NaN           2003.000000      NaN    105.000000    NaN   
75%             NaN           2008.000000      NaN    150.000000    NaN   
max             NaN           9999.000000      NaN  17700.000000    NaN   

         odometer  month_of_registration fuel_type       brand  \
count       50000           50000.000000     45518       50000   
unique         13                    NaN         7          40   
top     150,000km                    NaN    benzin  volkswagen   
freq        32424                    NaN     30107       10687   
mean          NaN               5.723360       NaN         NaN   
std           NaN               3.711984       NaN         NaN   
min           NaN               0.000000       NaN         NaN   
25%           NaN               3.000000       NaN         NaN   
50%           NaN               6.000000       NaN         NaN   
75%           NaN               9.000000       NaN         NaN   
max           NaN              12.000000       NaN         NaN   

       not_repaired_damage           ad_created  nr_of_pictures   postal_code  \
count                40171                50000         50000.0  50000.000000   
unique                   2                   76             NaN           NaN   
top                   nein  2016-04-03 00:00:00             NaN           NaN   
freq                 35232                 1946             NaN           NaN   
mean                   NaN                  NaN             0.0  50813.627300   
std                    NaN                  NaN             0.0  25779.747957   
min                    NaN                  NaN             0.0   1067.000000   
25%                    NaN                  NaN             0.0  30451.000000   
50%                    NaN                  NaN             0.0  49577.000000   
75%                    NaN                  NaN             0.0  71540.000000   
max                    NaN                  NaN             0.0  99998.000000   

                  last_seen  
count                 50000  
unique                39481  
top     2016-04-07 06:17:27  
freq                      8  
mean                    NaN  
std                     NaN  
min                     NaN  
25%                     NaN  
50%                     NaN  
75%                     NaN  
max                     NaN

Our initial observations:

- There are a number of text columns where all (or nearly all) of the values are the same:
    - `seller`
    - `offer_type`
- The `nr_of_pictures` column looks odd, we'll need to investigate this further.
- I don't see `postal_code` as an important variable in this analysis.
- `odometer` and `price` could be transformed in a numeric data.

In [15]:
autos["nr_of_pictures"].value_counts()

nr_of_pictures
0    50000
Name: count, dtype: int64

All values are 0 in `nr_of_pictures`', so it may be dropped.

In [17]:
autos = autos.drop(["nr_of_pictures", "seller", "offer_type"], axis=1) # dropping columns 

### Cleaning Price and Odometer columns

In [19]:
autos['odometer'] = (autos['odometer'].str.replace('km', '').str.replace(',','').astype('int'))

In [20]:
autos['price'] = (autos['price'].str.replace('$', '').str.replace(',','').astype('int'))

In [21]:
autos.rename({'odometer': 'odometer_km'}, axis = 1, inplace = True)
autos['odometer_km'].head()

0    150000
1    150000
2     70000
3     70000
4    150000
Name: odometer_km, dtype: int32

In [22]:
# looking for outliers
print(autos['odometer_km'].max(),
    autos['odometer_km'].min(),
    autos['price'].max(),
    autos['price'].min())

150000 5000 99999999 0


In [23]:
autos['price'].unique().shape

(2357,)

In [24]:
autos['price'].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

In [25]:
autos['price'].value_counts().sort_index(ascending = False).head(20) # highest prices

price
99999999    1
27322222    1
12345678    3
11111111    2
10000000    1
3890000     1
1300000     1
1234566     1
999999      2
999990      1
350000      1
345000      1
299000      1
295000      1
265000      1
259000      1
250000      1
220000      1
198000      1
197000      1
Name: count, dtype: int64

In [26]:
autos['odometer_km'].unique().shape

(13,)

In [27]:
autos['odometer_km'].describe()

count     50000.000000
mean     125732.700000
std       40042.211706
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

In [28]:
autos['odometer_km'].value_counts().sort_index(ascending = False).head(20)

odometer_km
150000    32424
125000     5170
100000     2169
90000      1757
80000      1436
70000      1230
60000      1164
50000      1027
40000       819
30000       789
20000       784
10000       264
5000        967
Name: count, dtype: int64

Founded some outliers in the 'price' column. I looked the rows of this outliers below. The outliers are Limousines or Nan values, so I decided to remove then from the data analysis, because the data is going to be cleaner without then and Limousines are a "different" type of car. The first car after the outliers had a bid price of $10.000.000.

In [30]:
result = autos[autos['price'] == 27322222]
result2 = autos[autos['price'] == 99999999]
result3 = autos[autos['price'] == 12345678]
result4 = autos[autos['price'] == 11111111]
result5 = autos[autos['price'] == 10000000]

print(result['vehicle_type'],'\n','\n') # looking just the variable of interest
print(result2['vehicle_type'],'\n','\n')

print(result3['vehicle_type'],'\n','\n')

print(result4['vehicle_type'],'\n','\n')

print(result5['vehicle_type'],'\nnn','\n')

42221    limousine
Name: vehicle_type, dtype: object 
 

39705    limousine
Name: vehicle_type, dtype: object 
 

27371          NaN
39377          NaN
47598    limousine
Name: vehicle_type, dtype: object 
 

2897     limousine
24384          NaN
Name: vehicle_type, dtype: object 
 

11137    coupe
Name: vehicle_type, dtype: object 
nn 



In [31]:
autos = autos[autos["price"].between(0,10000000)]
autos

date_crawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
...                    ...                                                ...   
49995  2016-03-27 14:38:19   Audi_Q5_3.0_TDI_qu._S_tr.__Navi__Panorama__Xenon   
49996  2016-03-28 10:50:25  Opel_Astra_F_Cabrio_Bertone_Edition___TÜV_neu+...   
49997  2016-04-02 14:44:48                     Fiat_500_C_1.2_Dualogic_Lounge   
49998  2016-03-08 19:25:42                 Audi_A3_2.0_TDI_Sportback_Ambition   
49999  2016-03-14 00:42:12                                Opel_Vectra_1.6_16V   

       price   abtest vehicle_type  year_of_registration    gearbox  power_ps  \
0       5000  control          bus                  2004    manuell       158   
1       8500  control    limousine                  1997  automatik       286   
2       8990     test    limousine                  2009    manuell       102   
3       4350  control   kleinwagen                  2007  automatik        71   
4       1350     test        kombi                  2003    manuell         0   
...      ...      ...          ...                   ...        ...       ...   
49995  24900  control    limousine                  2011  automatik       239   
49996   1980  control       cabrio                  1996    manuell        75   
49997  13200     test       cabrio                  2014  automatik        69   
49998  22900  control        kombi                  2013    manuell       150   
49999   1250  control    limousine                  1996    manuell       101   

        model  odometer_km  month_of_registration fuel_type       brand  \
0      andere       150000                      3       lpg     peugeot   
1         7er       150000                      6    benzin         bmw   
2        golf        70000                      7    benzin  volkswagen   
3      fortwo        70000                      6    benzin       smart   
4       focus       150000                      7    benzin        ford   
...       ...          ...                    ...       ...         ...   
49995      q5       100000                      1    diesel        audi   
49996   astra       150000                      5    benzin        opel   
49997     500         5000                     11    benzin        fiat   
49998      a3        40000                     11    diesel        audi   
49999  vectra       150000                      1    benzin        opel   

      not_repaired_damage           ad_created  postal_code  \
0                    nein  2016-03-26 00:00:00        79588   
1                    nein  2016-04-04 00:00:00        71034   
2                    nein  2016-03-26 00:00:00        35394   
3                    nein  2016-03-12 00:00:00        33729   
4                    nein  2016-04-01 00:00:00        39218   
...                   ...                  ...          ...   
49995                nein  2016-03-27 00:00:00        82131   
49996                nein  2016-03-28 00:00:00        44807   
49997                nein  2016-04-02 00:00:00        73430   
49998                nein  2016-03-08 00:00:00        35683   
49999                nein  2016-03-13 00:00:00        45897   

                 last_seen  
0      2016-04-06 06:45:54  
1      2016-04-06 14:45:08  
2      2016-04-06 20:15:37  
3      2016-03-15 03:16:28  
4      2016-04-01 14:38:50  
...                    ...  
49995  2016-04-01 13:47:40  
49996  2016-04-02 14:18:02  
49997  2016-04-04 11:47:27  
49998  2016-04-05 16:45:07  
49999  2016-04-06 21:18:48  

[49993 rows x 17 columns]

In [32]:
autos['price'].value_counts().sort_index(ascending = False).head(20)

price
10000000    1
3890000     1
1300000     1
1234566     1
999999      2
999990      1
350000      1
345000      1
299000      1
295000      1
265000      1
259000      1
250000      1
220000      1
198000      1
197000      1
194000      1
190000      1
180000      1
175000      1
Name: count, dtype: int64

### Analysing the year of registration and other important dates from the listed cars

In [34]:
autos[['date_crawled','ad_created','last_seen']][0:5]

date_crawled           ad_created            last_seen
0  2016-03-26 17:47:46  2016-03-26 00:00:00  2016-04-06 06:45:54
1  2016-04-04 13:38:56  2016-04-04 00:00:00  2016-04-06 14:45:08
2  2016-03-26 18:57:24  2016-03-26 00:00:00  2016-04-06 20:15:37
3  2016-03-12 16:58:10  2016-03-12 00:00:00  2016-03-15 03:16:28
4  2016-04-01 14:38:50  2016-04-01 00:00:00  2016-04-01 14:38:50

In [35]:
autos['date_crawled'].value_counts(normalize = True, dropna = False).sort_index() # normalize makes the series proportional

date_crawled
2016-03-05 14:06:30    0.00002
2016-03-05 14:06:40    0.00002
2016-03-05 14:07:04    0.00002
2016-03-05 14:07:08    0.00002
2016-03-05 14:07:21    0.00002
                        ...   
2016-04-07 14:30:09    0.00002
2016-04-07 14:30:26    0.00002
2016-04-07 14:36:44    0.00002
2016-04-07 14:36:55    0.00002
2016-04-07 14:36:56    0.00002
Name: proportion, Length: 48207, dtype: float64

In [36]:
autos['ad_created'].value_counts(normalize = True, dropna = False).sort_index()

ad_created
2015-06-11 00:00:00    0.000020
2015-08-10 00:00:00    0.000020
2015-09-09 00:00:00    0.000020
2015-11-10 00:00:00    0.000020
2015-12-05 00:00:00    0.000020
                         ...   
2016-04-03 00:00:00    0.038925
2016-04-04 00:00:00    0.036885
2016-04-05 00:00:00    0.011842
2016-04-06 00:00:00    0.003260
2016-04-07 00:00:00    0.001280
Name: proportion, Length: 76, dtype: float64

In [37]:
autos['last_seen'].value_counts(normalize = True, dropna = False).sort_index()

last_seen
2016-03-05 14:45:46    0.00002
2016-03-05 14:46:02    0.00002
2016-03-05 14:49:34    0.00002
2016-03-05 15:16:11    0.00002
2016-03-05 15:16:47    0.00002
                        ...   
2016-04-07 14:58:44    0.00006
2016-04-07 14:58:45    0.00002
2016-04-07 14:58:46    0.00002
2016-04-07 14:58:48    0.00006
2016-04-07 14:58:50    0.00008
Name: proportion, Length: 39477, dtype: float64

In [38]:
autos['year_of_registration'].describe()

count    49993.000000
mean      2005.073650
std        105.720065
min       1000.000000
25%       1999.000000
50%       2003.000000
75%       2008.000000
max       9999.000000
Name: year_of_registration, dtype: float64

**'year_of_registration' also have some issues with outliers as year = 9999 or = 1000.** A car can not be registered aftrer the listing was seen for the last time, thus, any car with a registration year above 2016 is inaccurate. The earliest valid year is more difficult. I will filter for cars that were registered after 1900.

In [40]:
autos = autos[autos["year_of_registration"].between(1900,2016)]
autos

date_crawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
...                    ...                                                ...   
49995  2016-03-27 14:38:19   Audi_Q5_3.0_TDI_qu._S_tr.__Navi__Panorama__Xenon   
49996  2016-03-28 10:50:25  Opel_Astra_F_Cabrio_Bertone_Edition___TÜV_neu+...   
49997  2016-04-02 14:44:48                     Fiat_500_C_1.2_Dualogic_Lounge   
49998  2016-03-08 19:25:42                 Audi_A3_2.0_TDI_Sportback_Ambition   
49999  2016-03-14 00:42:12                                Opel_Vectra_1.6_16V   

       price   abtest vehicle_type  year_of_registration    gearbox  power_ps  \
0       5000  control          bus                  2004    manuell       158   
1       8500  control    limousine                  1997  automatik       286   
2       8990     test    limousine                  2009    manuell       102   
3       4350  control   kleinwagen                  2007  automatik        71   
4       1350     test        kombi                  2003    manuell         0   
...      ...      ...          ...                   ...        ...       ...   
49995  24900  control    limousine                  2011  automatik       239   
49996   1980  control       cabrio                  1996    manuell        75   
49997  13200     test       cabrio                  2014  automatik        69   
49998  22900  control        kombi                  2013    manuell       150   
49999   1250  control    limousine                  1996    manuell       101   

        model  odometer_km  month_of_registration fuel_type       brand  \
0      andere       150000                      3       lpg     peugeot   
1         7er       150000                      6    benzin         bmw   
2        golf        70000                      7    benzin  volkswagen   
3      fortwo        70000                      6    benzin       smart   
4       focus       150000                      7    benzin        ford   
...       ...          ...                    ...       ...         ...   
49995      q5       100000                      1    diesel        audi   
49996   astra       150000                      5    benzin        opel   
49997     500         5000                     11    benzin        fiat   
49998      a3        40000                     11    diesel        audi   
49999  vectra       150000                      1    benzin        opel   

      not_repaired_damage           ad_created  postal_code  \
0                    nein  2016-03-26 00:00:00        79588   
1                    nein  2016-04-04 00:00:00        71034   
2                    nein  2016-03-26 00:00:00        35394   
3                    nein  2016-03-12 00:00:00        33729   
4                    nein  2016-04-01 00:00:00        39218   
...                   ...                  ...          ...   
49995                nein  2016-03-27 00:00:00        82131   
49996                nein  2016-03-28 00:00:00        44807   
49997                nein  2016-04-02 00:00:00        73430   
49998                nein  2016-03-08 00:00:00        35683   
49999                nein  2016-03-13 00:00:00        45897   

                 last_seen  
0      2016-04-06 06:45:54  
1      2016-04-06 14:45:08  
2      2016-04-06 20:15:37  
3      2016-03-15 03:16:28  
4      2016-04-01 14:38:50  
...                    ...  
49995  2016-04-01 13:47:40  
49996  2016-04-02 14:18:02  
49997  2016-04-04 11:47:27  
49998  2016-04-05 16:45:07  
49999  2016-04-06 21:18:48  

[48023 rows x 17 columns]

In [41]:
autos['year_of_registration'].value_counts(normalize = True)

year_of_registration
2000    0.069842
2005    0.062782
1999    0.062449
2004    0.056994
2003    0.056785
          ...   
1929    0.000021
1952    0.000021
1939    0.000021
1938    0.000021
1953    0.000021
Name: proportion, Length: 78, dtype: float64

In [42]:
autos['year_of_registration'].describe()

count    48023.000000
mean      2002.804177
std          7.309675
min       1910.000000
25%       1999.000000
50%       2003.000000
75%       2008.000000
max       2016.000000
Name: year_of_registration, dtype: float64

## Analysing the Top Car Brands in Ebay

In [44]:
autos['brand'].unique()

array(['peugeot', 'bmw', 'volkswagen', 'smart', 'ford', 'chrysler',
       'seat', 'renault', 'mercedes_benz', 'audi', 'sonstige_autos',
       'opel', 'mazda', 'porsche', 'mini', 'toyota', 'dacia', 'nissan',
       'jeep', 'saab', 'volvo', 'mitsubishi', 'jaguar', 'fiat', 'skoda',
       'subaru', 'kia', 'citroen', 'chevrolet', 'hyundai', 'honda',
       'daewoo', 'suzuki', 'trabant', 'land_rover', 'alfa_romeo', 'lada',
       'rover', 'daihatsu', 'lancia'], dtype=object)

In [45]:
brand_counts = autos["brand"].value_counts(normalize=True)
common_brands = brand_counts[brand_counts > .05].index
print(common_brands)

Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford'], dtype='object', name='brand')


I have choosen the top 6 brands in Ebay to use for the analysis.

In [47]:
brand_mean_prices = {} # This dict wll be usefull to create a DF later

for i in common_brands:

    brand_only = autos[autos['brand'] == i]
    mean_price = brand_only['price'].mean()
    brand_mean_prices[i] = int(mean_price)

brand_mean_prices

{'volkswagen': 5426,
 'bmw': 8334,
 'opel': 2876,
 'mercedes_benz': 8485,
 'audi': 9093,
 'ford': 3949}

In [48]:
brand_mean_km = {}

for i in common_brands:

    brand_only = autos[autos['brand'] == i]
    mean_km = brand_only['odometer_km'].mean()
    brand_mean_km[i] = int(mean_km)

brand_mean_km

{'volkswagen': 128728,
 'bmw': 132434,
 'opel': 129223,
 'mercedes_benz': 130856,
 'audi': 129287,
 'ford': 124068}

In [49]:
brand_mean_price_series = pd.Series(brand_mean_prices)
brand_mean_km_series = pd.Series(brand_mean_km)

In [50]:
autos_df = pd.DataFrame(brand_mean_price_series, columns = ['mean_price'])

In [51]:
autos_df['mean_km'] = brand_mean_km

In [52]:
autos_df

mean_price  mean_km
volkswagen           5426   128728
bmw                  8334   132434
opel                 2876   129223
mercedes_benz        8485   130856
audi                 9093   129287
ford                 3949   124068

**Looking the data, there is a little positive correlation between mean price and mean_km. BMW, Mercedes and Audi are clearly more expensive because they are luxury vehicles.**

In [54]:
autos_df['mean_price'].corr(autos_df['mean_km'])

0.616199355232792

## Finding the most common brand/model combinations

In [56]:
autos['model'].describe()

count     45556
unique      244
top        golf
freq       3815
Name: model, dtype: object

In [57]:
model_types = autos['model'].value_counts(ascending = False)[:10]
print(model_types)

model
golf        3815
andere      3457
3er         2688
polo        1677
corsa       1645
astra       1388
passat      1388
a4          1265
5er         1163
c_klasse    1147
Name: count, dtype: int64


In [58]:
# Group by brand and model, then count occurrences
brand_model_counts = autos.groupby(['brand', 'model']).size().reset_index(name='count')

# Sort by count in descending order
brand_model_counts = brand_model_counts.sort_values('count', ascending=False)

# Display the top N results (e.g., top 10)
top_n = 10
print(brand_model_counts.head(top_n))

             brand     model  count
267     volkswagen      golf   3815
23             bmw       3er   2688
273     volkswagen      polo   1677
182           opel     corsa   1645
179           opel     astra   1388
271     volkswagen    passat   1388
13            audi        a4   1265
24             bmw       5er   1163
141  mercedes_benz  c_klasse   1147
144  mercedes_benz  e_klasse    981


## How much cheaper are cars with damage than their non-damaged counterparts?

In [60]:
damaged_count = autos["not_repaired_damage"].value_counts()
damaged = damaged_count.index
print(damaged_count)

not_repaired_damage
nein    34253
ja       4785
Name: count, dtype: int64


In [61]:
# Calculate mean prices for damaged and non-damaged cars
mean_price_not_damaged = autos[autos['not_repaired_damage'] == 'nein']['price'].mean()
mean_price_damaged = autos[autos['not_repaired_damage'] == 'ja']['price'].mean()

In [62]:
# Calculate the price difference
price_difference = mean_price_not_damaged - mean_price_damaged
print(price_difference)

5213.678393353093


In [63]:
# Calculate median prices to account for potential outliers
median_price_not_damaged = autos[autos['not_repaired_damage'] == 'nein']['price'].median()
median_price_damaged = autos[autos['not_repaired_damage'] == 'ja']['price'].median()
median_difference = median_price_not_damaged - median_price_damaged

In [64]:
print(median_difference)

3035.0


## Conclusions

* The average car of EBAY in Germany is from 2002 in the dataset.
* 'volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford' are the TOP Brands in German Ebay.
* There is a positive correlation between `mean_price` and  `mean_km`, but correlation does not mean causality.
* The most popular car for listings in EBAY is Golf from Volkswagen.
* **The median  damaged car is 3,000 euros cheaper then the median not damaged car**